app/src/components/mode-specific/desktop/InterceptTraffic/WebTraffic/TrafficTableV2/Preview/JsonPreviewV2.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useMemo } from "react";
+import { ThemeProvider } from "@devtools-ds/themes";
+import { ObjectInspector } from "@devtools-ds/object-inspector";
+import { Typography } from "antd";
+
+const MAX_ALLOWED_NETWORK_RESPONSE_SIZE = 200 * 1024; // 200KB
+const isWithInLimit = (data: any) => {
+  try {
+    if (typeof data == "object") {
+      return JSON.stringify(data).length < MAX_ALLOWED_NETWORK_RESPONSE_SIZE;
+    }
+    if (typeof data == "string") {
+      return data.length < MAX_ALLOWED_NETWORK_RESPONSE_SIZE;
+    }
+  } catch (e) {
+    // todo: can add custom error to show different text for unsupported preview
+  }
+
+  return false;
+};
+
+type Props = {
+  payload: any;
+  logId: string;
+};
+
+const JSONPreview: React.FC<Props> = ({ payload, logId }) => {
+  const parsedPayload = useMemo(() => {
+    try {
+      if (isWithInLimit(payload)) {
+        if (typeof payload === "string") {
+          const parsedObject = JSON.parse(payload);
+          return parsedObject;
+        }
+        return payload;
+      }
+      return "Response too large / Not supported";
```

Comment:
I don't see when this is visible in UI. Also, this should be wrapped inside a secondary typography text.

Reply:
This will be shown when the stringified version of the payload exceeds the set hard limit. Eyeballed this to be 200KB for now, cause we also sometimes have byte arrays as responses (this is due to a known encoding issue inside the requestly proxy)